web/static/js/components/thesis-editor.js

```diff
@@ -223,6 +264,44 @@ class ThesisEditor extends React.Component {
     }
   }
 
+  importContent (t, d, page) {
+    switch (t) {
+      case 'image':
+      case 'background_image':
+        return this.imageEditor.uploadAndSet(d, page, () => { this.updateImportedContentCompletedCount() })
+      case 'text':
+        this.updateImportedContentCompletedCount()
+        return this.textEditor.set(d.name, d)
+      case 'html':
+        this.updateImportedContentCompletedCount()
+        return this.htmlEditor.set(d.name, d)
+      case 'raw_html':
+        this.updateImportedContentCompletedCount()
+        return this.rawHtmlEditor.set(d.name, d)
+      default:
+        return true
+    }
+  }
+
+  updateImportedContentCompletedCount () {
+    this.setState({importContentCompletedCount: ++this.state.importContentCompletedCount}, () => {
+      if (this.state.importContentCompletedCount === this.state.importContentQueueCount) {
+        clearTimeout(window.importContentCounterTimeout)
+
+        this.setState({
+          importContentQueueCount: 0,
+          importContentCompletedCount: 0,
+          importProgress: null,
+          trayOpen: false
+        })
+      }
+    })
+  }
```

Comment:
this is the logic to determine when all the content has been updated. when the completed matches the initial queue, we clear the error timeout and finalize the import state